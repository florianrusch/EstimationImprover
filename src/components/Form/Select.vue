<template>
	<div>
		<select
			class="
				outline-none
				focus:outline-none
				border-b
				px-3
				py-1
				bg-white
				rounded-sm
				flex
				items-center
				min-w-32
			"
			@change="onChange"
		>
			<option
				v-for="ele in options"
				:key="ele.name"
				:value="ele.value"
				:selected="current == ele.value"
			>
				{{ ele.name }}
			</option>
		</select>
	</div>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue'

interface Option {
	name: string
	value: number
}

export default defineComponent({
	props: {
		current: {
			type: Number,
			required: true,
		},
		options: {
			type: Array as PropType<Option[]>,
			required: true,
		},
	},

	emits: ['change'],

	setup(props, { emit }) {
		// eslint-disable-next-line @typescript-eslint/no-explicit-any
		const onChange = (event: any) => {
			emit('change', event.target.value)
		}
		return {
			onChange,
		}
	},
})
</script>
