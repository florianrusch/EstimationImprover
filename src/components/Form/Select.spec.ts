import { DOMWrapper, mount } from '@vue/test-utils'
import Select from './Select.vue'

const mockOptions = [
	{ name: 'Test1', value: 0 },
	{ name: 'Test2', value: 1 },
]

describe('# Select', () => {
	it('renders correctly w/o init value', () => {
		// given
		const wrapper = mount(Select, {
			props: {
				options: mockOptions,
			},
		})

		// when
		// then
		expect(wrapper.element).toMatchSnapshot()
	})

	it('renders correctly w/ init value', () => {
		// given
		const wrapper = mount(Select, {
			props: {
				current: 0,
				options: mockOptions,
			},
		})

		// when
		// then
		expect(wrapper.element).toMatchSnapshot()
	})

	it('emits change event with new value', async () => {
		// given
		const wrapper = mount(Select, {
			props: {
				current: 0,
				options: mockOptions,
			},
		})

		// when
		await (wrapper.find('select') as DOMWrapper<HTMLSelectElement>).setValue(mockOptions[1].value)

		// then
		const events = wrapper.emitted('change')
		expect(events).toHaveLength(1)
		expect(events?.[0]).toEqual([mockOptions[1].value.toString()])
	})
})
