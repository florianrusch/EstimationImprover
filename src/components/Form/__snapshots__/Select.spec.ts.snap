// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Select renders correctly w/ init value 1`] = `
<div>
  <select
    class="
				outline-none
				focus:outline-none
				border-b
				px-3
				py-1
				bg-white
				rounded-sm
				flex
				items-center
				min-w-32
			"
  >
    <!--v-if-->
    
    <option
      value="0"
    >
      Test1
    </option>
    <option
      value="1"
    >
      Test2
    </option>
    
  </select>
</div>
`;

exports[`# Select renders correctly w/o init value 1`] = `
<div>
  <select
    class="
				outline-none
				focus:outline-none
				border-b
				px-3
				py-1
				bg-white
				rounded-sm
				flex
				items-center
				min-w-32
			"
  >
    <option
      disabled=""
    />
    
    <option
      value="0"
    >
      Test1
    </option>
    <option
      value="1"
    >
      Test2
    </option>
    
  </select>
</div>
`;
